tests/Tests.FeatureManagement/RecurrenceEvaluator.cs

```diff
@@ -0,0 +1,2429 @@
+                        {
+                            Type = "NoEnd"
+                        }
+                    }
+                },
+                true ),
+
+                ( DateTimeOffset.Parse("2023-10-27T00:00:00+08:00"), // 4th Friday in 2023 Oct
+                new TimeWindowFilterSettings()
+                {
+                    Start = DateTimeOffset.Parse("2023-9-29T00:00:00+08:00"),
+                    End = DateTimeOffset.Parse("2023-9-29T00:00:01+08:00"),
+                    Recurrence = new Recurrence()
+                    {
+                        Pattern = new RecurrencePattern()
+                        {
+                            Type = "RelativeMonthly",
+                            DaysOfWeek = new List<string>() { "Friday" },
+                            Index = "Last"
+                        },
+                        Range = new RecurrenceRange()
+                        {
+                            Type = "NoEnd"
+                        }
+                    }
+                },
+                true ),
+
+                ( DateTimeOffset.Parse("2023-11-24T00:00:00+08:00"), // 4th Friday in 2023 Nov
+                new TimeWindowFilterSettings()
+                {
+                    Start = DateTimeOffset.Parse("2023-9-29T00:00:00+08:00"),
+                    End = DateTimeOffset.Parse("2023-9-29T00:00:01+08:00"),
+                    Recurrence = new Recurrence()
+                    {
+                        Pattern = new RecurrencePattern()
+                        {
+                            Type = "RelativeMonthly",
+                            DaysOfWeek = new List<string>() { "Friday" },
+                            Index = "Last"
+                        },
+                        Range = new RecurrenceRange()
+                        {
+                            Type = "NoEnd"
+                        }
+                    }
+                },
+                true ),
+
+                ( DateTimeOffset.Parse("2023-12-29T00:00:00+08:00"), // 5th Friday in 2023 Dec
+                new TimeWindowFilterSettings()
+                {
+                    Start = DateTimeOffset.Parse("2023-9-29T00:00:00+08:00"),
+                    End = DateTimeOffset.Parse("2023-9-29T00:00:01+08:00"),
+                    Recurrence = new Recurrence()
+                    {
+                        Pattern = new RecurrencePattern()
+                        {
+                            Type = "RelativeMonthly",
+                            DaysOfWeek = new List<string>() { "Friday" },
+                            Index = "Last"
+                        },
+                        Range = new RecurrenceRange()
+                        {
+                            Type = "NoEnd"
+                        }
+                    }
+                },
+                true ),
+
+                ( DateTimeOffset.Parse("2023-10-29T00:00:00+08:00"), // 4th Sunday in 2023 Oct
```

Comment:
"2023-10-29" should be the 5th Sunday in 2023 Oct?
1st Sunday: 2023-10-01
2nd Sunday: 2023-10-08
3rd Sunday: 2023-10-15
4th Sunday: 2023-10-22
5th Sunday: 2023-10-29